robot/impl/local_robot_test.go

```diff
@@ -3134,19 +3182,468 @@ func TestMachineStatus(t *testing.T) {
 		// now all components, including the one whose config changed, should all be in
 		// the "ready" state and associated with the new revision.
 		expectedStatuses := rtestutils.ConcatResourceStatuses(
-			getExpectedDefaultStatuses(rev2),
+			getExpectedDefaultStatuses(rev2, cloud.Metadata{}),
 			[]resource.Status{
 				{
-					Name:     mockNamed("m"),
-					State:    resource.NodeStateReady,
-					Revision: rev2,
+					NodeStatus: resource.NodeStatus{
+						Name:     mockNamed("m"),
+						State:    resource.NodeStateReady,
+						Revision: rev2,
+					},
 				},
 			},
 		)
 		rtestutils.VerifySameResourceStatuses(t, mStatus.Resources, expectedStatuses)
 	})
 }
 
+func TestMachineStatusWithRemotes(t *testing.T) {
```

Comment:
FYI I think the three tests you've added here will break with my `MachineStatus` PR; just noting for whomever happens to merge first. Shouldn't be a huge fix; just making sure `State` with `robot.StateRunning` is included in any injected `MachineStatusFunc` returns.